### DISCUSSION TITLE
Why `@tauri-apps/plugin-fs` is so much slower than File System API

### DISCUSSION THREAD

**Discussion by DexYang** [tauri-apps/tauri#9700] [Q&A] [answered]
Why `@tauri-apps/plugin-fs` is so much slower than [File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API)

I load the same files, `@tauri-apps/plugin-fs` is 100x slower ....

 `@tauri-apps/plugin-fs`  ==>>
```
import { readFile } from '@tauri-apps/plugin-fs'
const buf= await readFile(<path>)
```
![1715148683964](https://github.com/tauri-apps/tauri/assets/22140560/cccdf7e4-5d44-41c5-9dd7-cdcb76fc8863)

[File System API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API)  ==>>

```
const fileHandle = await <FileSystemHandle>.getFile()
const buf = await fileHandle.slice(start, end).arrayBuffer()
```
![1715148677150](https://github.com/tauri-apps/tauri/assets/22140560/4e9e79bd-ed31-4a3a-a696-3acfb656dae5)

**Comment by FabianLars** [comment, marked as answer]
Because the IPC, the communication channel between rust (where the plugin-fs is actually implemented) and js is pretty slow. We've vastly improved the performance in v2 compared to v1 but there are still a few things we have to fix, for example https://github.com/tauri-apps/plugins-workspace/pull/1251 should help a ton already.

**Reply by DexYang** [reply]
thx

seems fs still not using the IPC？looking forward the performance improving😀
